pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+		},
+		Data: map[string][]byte{
+			"id_rsa":     []byte(privkey),
+			"id_rsa.pub": []byte(pubkey),
+		},
+	}
+
+	return svc.SetDesiredKubeObject(secret, objName, runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createServiceAccount(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-sa",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+	}
+	return svc.SetDesiredKubeObject(sa, comp.GetName()+"-collabora-code-sa")
+	// return SetDesiredKubeObject.SetDesiredKubeObserveObjectWithLabels(sa, comp.GetName()+"-collabora-code-sa", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createSecurityContextRole(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-role",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				// pods
+				APIGroups: []string{""},
+				Resources: []string{"pods", "pods/log", "pods/status", "pods/attach", "pods/exec"},
+				Verbs:     []string{"*"},
+			},
+			{
+				// deployments
+				APIGroups: []string{"apps"},
+				Resources: []string{"deployments", "statefulsets"},
+				Verbs:     []string{"get", "delete", "watch", "list", "patch", "update", "create"},
+			},
+		},
+	}
+
+	if svc.Config.Data["isOpenshift"] == "true" {
+		role.Rules = append(role.Rules, rbacv1.PolicyRule{
+			APIGroups:     []string{"security.openshift.io"},
+			Resources:     []string{"securitycontextconstraints"},
+			ResourceNames: []string{"appcat-collabora"},
+			Verbs:         []string{"use"},
+		})
+	}
+
+	return svc.SetDesiredKubeObject(role, comp.GetName()+"-collabora-code-role", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createSecurityContextRoleBinding(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-role-binding",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      comp.GetName() + "-collabora-code-sa",
+				Namespace: comp.GetInstanceNamespace(),
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     comp.GetName() + "-collabora-code-role",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	return svc.SetDesiredKubeObject(roleBinding, comp.GetName()+"-collabora-code-role-binding", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createWatchOnlyObject(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	// Create a watch-only object to trigger a re-deploy
```

Comment:
I'm a bit confused about the purpose of this secret. What should get re-deployed?

Reply:
Yes I am also puzzled what is happening here. In func `createSecretForOpenshiftRoute()` this secret is referenced as watch but no idea what id does.

Reply:
Ah this is pretty simple:
1) I create TLS certificates via certmanager
2) Openshift Route requires that `ca.crt` is named `tls.crt` in Secret{} to ensure `reencrypt`.   

In another function I take values from it and assign to new Secret{} so Openshift Route is happy: https://github.com/vshn/appcat/pull/245/files#diff-505d860e0dbed16ff5b9ce081f5cfb7d5c6f62ef51d965917d683c157221d8fdR577

I'll add verbose comment about that

Reply:
Thanks, by renaming it to watch-only alone it's now clear.